### PR TITLE
feat: locale picker in the storybook's decorator

### PR DESCRIPTION
**Pull request by mizgaionutalexandru**
## Description
This PR adds a new control in the storybook decorator that helps the user change the language/locale.
In order to leverage the interaction of `sp-theme` and `LanguageResolutionController` in the storybook it is needed to use a workaround similar to `requestAnimationFrame` so that the story functions get called after the `Theme`'s constructor. 

For more context please check out [this](https://github.com/adobe/spectrum-web-components/pull/4667#discussion_r1713856107) PR comment.
## Motivation and context

Currently if a component is reactive to locale/language changes, in order for the storybook to reflect that it is needed for multiple stories to be created. This doesn't scale as well and it's also cumbersome for users of the SWC storybook to clearly identify the differences between locales.
Currently the most notable behavior change is in the `sp-number-field` component.

## How has this been tested?

-   [ ] _Test case 1_
    1. Go to `sp-number-field` [storybook](https://imizga-locale-story-decorator--spectrum-web-components.netlify.app/storybook/?path=/story/number-field--default)
    2. Type in a number that has group separators and decimals 
    3. Observe how it is shown with the en-US locale (e.g. 1,000.56)
    4. Change the locale to fr-FR 
    5. Observe the new formatted value (e.g. 1 000,56)
-   [ ] _Test case 1_
    1. Go to `sp-number-field` [storybook](https://imizga-locale-story-decorator--spectrum-web-components.netlify.app/storybook/?path=/story/number-field--default)
    2. Type in a number that has group separators and decimals 
    3. Observe how it is shown with the en-US locale (e.g. 1,000.56)
    4. Change the locale to de-DE 
    5. Observe the new formatted value (e.g. 1.000,56)

-   [x] Did it pass in Desktop?
-   [ ] Did it pass in Mobile?
-   [ ] Did it pass in iPad?

## Screenshots (if appropriate)

Screenshot 2024-08-23 at 18 03 26

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Comment by mizgaionutalexandru** [comment]
It looks like on iPad and mobile the user has a hard time using these controls from the Storybook decorator. Should we place some of them in a Action Menu?

File: projects/story-decorator/src/locales.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    'nl-NL',
+    'pl-PL',
+    'pt-BR',
+    'ru-RU',
+    'sv-SE',
+    'tr-TR',
+    'uk-UA',
+    'zh-Hans-CN',
+    'zh-Hant-TW',
+] as const;
```
Comment (mizgaionutalexandru) [inline projects/story-decorator/src/locales.ts:37]: I'm open to suggestions about this list of locales. Either to add or remove entries

Comment (rubencarvalho) [reply projects/story-decorator/src/locales.ts:37]: Do you know if this has parity with our most common internal use-cases?

**Comment by rubencarvalho** [comment]
There seems to be an inconsistent behavior: when you toggle "Reduce motion," the picker displays "Locale" instead of the selected value. Could you please check what's causing this? :)

https://github.com/user-attachments/assets/a2172801-d992-4d36-a86e-7d8ae35e630a

**Review by rubencarvalho — CHANGES_REQUESTED** [review]

File: projects/story-decorator/src/locales.ts (hunk 1 above)
Comment (mizgaionutalexandru) [reply projects/story-decorator/src/locales.ts]: I checked out [@intl/adobe-locales](https://git.corp.adobe.com/intl/adobe-locales#react-spectrum) and found out that ReactSpectrum supports 32 locales, though in the [source code](https://github.com/adobe/react-spectrum/blob/main/.storybook/constants.js#L66) I've found slightly more supported (37).
I've updated the list with a list of 27 locales with their code and language as per [this list](https://git.corp.adobe.com/intl/adobe-locales#projectx).

**Comment by mizgaionutalexandru** [comment]
> There seems to be an inconsistent behavior: when you toggle "Reduce motion," the picker displays "Locale" instead of the selected value. Could you please check what's causing this? :)

Thank you for looking into this. Missed a small `break` in the switch case. It should be fixed now. 🚀

File: projects/story-decorator/src/locales.ts (hunk 1 above)
Comment (rubencarvalho) [reply projects/story-decorator/src/locales.ts]: Thank you for looking into this :)

**Review by rubencarvalho — APPROVED** [review]
Great work! 👍 Let's 🚢 it.

**Pull request merged**